test/framework/deployment_helpers.go

```diff
@@ -122,7 +123,9 @@ func WatchDeploymentLogs(ctx context.Context, input WatchDeploymentLogsInput) {
 
 	deployment := &appsv1.Deployment{}
 	key := client.ObjectKeyFromObject(input.Deployment)
-	Expect(input.GetLister.Get(ctx, key, deployment)).To(Succeed(), "Failed to get deployment %s/%s", input.Deployment.Namespace, input.Deployment.Name)
+	Eventually(func() error {
+		return input.GetLister.Get(ctx, key, deployment)
+	}, retryableOperationTimeout, retryableOperationInterval).Should(Succeed(), "Failed to get deployment %s/%s", input.Deployment.Namespace, input.Deployment.Name)
```

Comment:
Unrelated directly to this PR but an observation:

Shouldn't this `Eventually` (and also at the old file the `Except` have an `.Should(Succeed())` assertion at the end?

Reply:
ah missed that, sorry 👍